### PR TITLE
#799 rest of Checked* classes

### PR DESCRIPTION
**Pull request by Vatavuk**
Introduced `Checked*` classes for #799

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#842](https://github.com/yegor256/cactoos/issues/842) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#842](https://github.com/yegor256/cactoos/issues/842) is assigned to @proshin-roman[/z](https://www.0crat.com/u/proshin-roman), here is [why](https://www.0crat.com/footprint/C63314D6Z/1282057808); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/main/java/org/cactoos/func/CheckedBiFunc.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * SOFTWARE.
+ */
+package org.cactoos.func;
+
+import org.cactoos.BiFunc;
+import org.cactoos.Func;
+import org.cactoos.scalar.CheckedScalar;
+
+/**
+ * BiFunc that throws exception of specified type.
```
Comment (proshin-roman) [inline src/main/java/org/cactoos/func/CheckedBiFunc.java]: @Vatavuk please, add an empty line after `BiFunc that throws exception of specified type.`

File: src/main/java/org/cactoos/func/CheckedBiProc.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ * SOFTWARE.
+ */
+package org.cactoos.func;
+
+import org.cactoos.BiProc;
+import org.cactoos.Func;
+import org.cactoos.scalar.CheckedScalar;
+
+/**
+ * BiProc that throws exception of specified type.
```
Comment (proshin-roman) [inline src/main/java/org/cactoos/func/CheckedBiProc.java]: @Vatavuk here too: please add an empty line

File: src/main/java/org/cactoos/func/CheckedFunc.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.func;
+
+import org.cactoos.Func;
+import org.cactoos.scalar.CheckedScalar;
+
+/**
+ * Func that throws exception of specified type.
```
Comment (proshin-roman) [inline src/main/java/org/cactoos/func/CheckedFunc.java]: @Vatavuk here too: add an empty line

File: src/main/java/org/cactoos/func/CheckedProc.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+ * SOFTWARE.
+ */
+package org.cactoos.func;
+
+import org.cactoos.Func;
+import org.cactoos.Proc;
+import org.cactoos.scalar.CheckedScalar;
+
+/**
+ * Proc that throws exception of specified type.
```
Comment (proshin-roman) [inline src/main/java/org/cactoos/func/CheckedProc.java]: @Vatavuk and here too: please add an empty line

File: src/test/java/org/cactoos/func/CheckedBiProcTest.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+import java.io.EOFException;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsEqual;
+import org.hamcrest.core.IsNull;
+import org.junit.Test;
+
+/**
+ * Test case for {@link CheckedBiProc}.
```
Comment (proshin-roman) [inline src/test/java/org/cactoos/func/CheckedBiProcTest.java]: @Vatavuk please add an empty line

**Comment by Vatavuk** [comment]
@proshin-roman fixed and merged with master

**Comment by Vatavuk** [comment]
@proshin-roman I don't know why there is no codecov report but it should be 100% because I've added tests that don't throw exception. Throwing exceptions in tests seems to be a problem for jaccoco.

**Comment by proshin-roman** [comment]
@Vatavuk sorry, but I - as a reviewer - can't approve the pull request without looking at the test coverage report. I don't know exactly how to fix this problem. Maybe you can find more [here](https://docs.codecov.io/docs/error-reference#section-missing-base-commit). My current assumption is that your branch is based on a commit, that was failing to build. Please, try to solve this problem.

**Comment by proshin-roman** [comment]
@Vatavuk I've submitted a bug https://github.com/yegor256/cactoos/issues/846 about the problem

**Comment by proshin-roman** [comment]
@llorllale @vatavuk it seems that the feature branch should be rebased so that its base commit has a test coverage report

**Comment by llorllale** [comment]
@proshin-roman I believe Travis builds pull requests by merging source+target branches. Unfortunately, each travis PR build is associated with a specific commit, so if @Vatavuk updates his branch with `master` it should create a *new* Travis build with the [fix](fc157ce6941a0881faa9d11256bea485654172bc).

**Comment by llorllale** [comment]
@proshin-roman I'm looking through other fresh PRs and I now suspect you're right.

@Vatavuk can you please try rebasing to 57a3c18175ef35e79a1bd1c35284be7db1c228b5 ?

**Comment by Vatavuk** [comment]
@llorllale rebased, seems fine now

**Comment by Vatavuk** [comment]
@proshin-roman coverage is good

**Comment by proshin-roman** [comment]
@rultor merge
@Vatavuk thanks for the fixes, PR is perfect now

**Comment by proshin-roman** [comment]
@rultor merge

**Comment by llorllale** [comment]
@rultor merge

**Comment by llorllale** [comment]
@Vatavuk #794 just went into `master` - you're going to have to remove `@author`, `@version`, and the license comment must start with `/*` (a single asterisk)

**Comment by Vatavuk** [comment]
@llorllale I've removed tags, please try again

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @proshin-roman[/z](https://www.0crat.com/u/proshin-roman), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#842](https://github.com/yegor256/cactoos/issues/842) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by elenavolokhova** [comment]
@proshin-roman According to our Policy:
> Comments were mostly about design problems, not cosmetic issues;

Please, confirm that you'll try to find at least three not cosmetic problems in the code next time.

**Comment by proshin-roman** [comment]
@elenavolokhova confirm

**Comment by elenavolokhova** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @proshin-roman[/z](https://www.0crat.com/u/proshin-roman)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Pull request merged**